### PR TITLE
Catch info from metadata

### PR DESCRIPTION
**Pull request by ManfeiBai**
Update code to catch info from metadata

### CODE REVIEW

File: torch_xla/experimental/tpu.py [hunk 1]
```diff
@@ -172,7 +172,7 @@ def configure_topology(local_rank: int,
   tpu_env = get_tpu_env()
 
   accelerator_type = tpu_env[xenv.ACCELERATOR_TYPE]
-  if version() == 4:
+  if tpu_env['ACCELERATOR_TYPE'].startswith('v4') or tpu_env['ACCELERATOR_TYPE'].startswith('v5'):
```
Comment (JackCaoG) [inline torch_xla/experimental/tpu.py:175]: @will-cromar what's the right place to add a test for this one? I felt like we had similar tests somwehre.

Comment (will-cromar) [reply torch_xla/experimental/tpu.py:175]: The relevant unit test is here: https://github.com/pytorch/xla/blob/18674daf0c44bef0162cbeddaaef7acc206ecea5/test/pjrt/test_experimental_tpu.py#L90

Can we just do `tpu.version() >= 4` here?

Comment (ManfeiBai) [reply torch_xla/experimental/tpu.py:175]: tried version() >=4, failed with https://paste.googleplex.com/5365404993060864

File: torch_xla/experimental/tpu.py [hunk 2]
```diff
@@ -127,7 +127,7 @@ def version() -> int:
   except requests.HTTPError as e:
     raise EnvironmentError('Failed to get TPU metadata') from e
 
-  match = re.match(r'^v(\d)([A-Za-z]?){7}-(\d+)$', env[xenv.ACCELERATOR_TYPE])
```
Comment (suexu1025) [inline torch_xla/experimental/tpu.py:130]: need to be `([A-Za-z]?){7}-(\d+)$` according to previous change # 96fd3ce

**Review by suexu1025 — CHANGES_REQUESTED** [review]
one question regarding #ACCELERATOR_TYPE match

File: torch_xla/experimental/tpu.py (hunk 2 above)
Comment (ManfeiBai) [reply torch_xla/experimental/tpu.py:130]: r'^v(\d)([A-Za-z]?){7}-(\d+)$' could match v0abcdefg;

and we also have v0abcd;

r'^v(\d).*' match v0abcd and v0abcdefg

Comment (ManfeiBai) [reply torch_xla/experimental/tpu.py:130]: maybe we could try r'^v(\d).*-(\d+)$'

**Review by yeounoh — APPROVED** [review]
LGTM

File: torch_xla/experimental/tpu.py [hunk 3]
```diff
@@ -127,8 +127,7 @@ def version() -> int:
...
-  return int(match.groups()[0])
+  return int((env[xenv.ACCELERATOR_TYPE])[1])
```
Comment (will-cromar) [inline torch_xla/experimental/tpu.py:130]: IMO the regex here is clearer. Do you have a reason to change it? The existing one works in all of the test cases:

```
>>> re.match(r'^v(\d)([A-Za-z]?){7}-(\d+)$', 'v5abcdefg-4').groups()[0]
'5'
>>> re.match(r'^v(\d)([A-Za-z]?){7}-(\d+)$', 'v5abcd-4').groups()[0]
'5'
>>> re.match(r'^v(\d)([A-Za-z]?){7}-(\d+)$', 'v5-4').groups()[0]
'5'
```

**Review by will-cromar — CHANGES_REQUESTED** [review]

File: torch_xla/experimental/tpu.py (hunk 2 above)
Comment (ManfeiBai) [reply torch_xla/experimental/tpu.py:130]: updated to `r'^v(\d).*' ` to match `v0abcdefg-higklmnop-0`

File: torch_xla/experimental/tpu.py (hunk 3 above)
Comment (ManfeiBai) [reply torch_xla/experimental/tpu.py]: Thanks, updated

Comment (ManfeiBai) [reply torch_xla/experimental/tpu.py]: Thanks, Will @will-cromar , after the modification based on your advice, current code has passed the CI tests too, do we want to mark this request change as finished and submit this PR?

**Review by will-cromar — APPROVED** [review]
Thanks!

**Pull request merged**